workflows/ascc.nf

```diff
@@ -203,83 +240,275 @@ workflow ASCC {
+    if ( include_workflow_steps.contains('vecscreen') || include_workflow_steps.contains('ALL') ) {
         RUN_VECSCREEN (
             GENERATE_GENOME.out.reference_tuple,
             YAML_INPUT.out.vecscreen_database_path
         )
-        ch_versions = ch_versions.mix(RUN_VECSCREEN.out.versions)
+        ch_vecscreen    = RUN_VECSCREEN.out.vecscreen_contam.map{it[1]}
+        ch_versions     = ch_versions.mix(RUN_VECSCREEN.out.versions)
+    } else {
+        ch_vecscreen    = []
     }
 
     //
     // SUBWORKFLOW: Run the kraken classifier
     //
-    if ( workflow_steps.contains('kraken') || workflow_steps.contains('ALL') ) {
+    if ( include_workflow_steps.contains('kraken') || include_workflow_steps.contains('ALL') ) {
         RUN_NT_KRAKEN(
             GENERATE_GENOME.out.reference_tuple,
             YAML_INPUT.out.nt_kraken_db_path,
             YAML_INPUT.out.ncbi_rankedlineage_path
         )
+        ch_kraken1      = RUN_NT_KRAKEN.out.classified.map{it[1]}
+        ch_kraken2      = RUN_NT_KRAKEN.out.report.map{it[1]}
+        ch_kraken3      = RUN_NT_KRAKEN.out.lineage
+
+        ch_versions     = ch_versions.mix(RUN_NT_KRAKEN.out.versions)
+    } else {
+        ch_kraken1      = []
+        ch_kraken2      = []
+        ch_kraken3      = []
+    }
+
+    //
+    // SUBWORKFLOW: DIAMOND BLAST FOR INPUT ASSEMBLY
+    //
+    if ( include_workflow_steps.contains('nt_diamond') || include_workflow_steps.contains('ALL') ) {
+        NUCLEOT_DIAMOND (
+            modified_input,
+            YAML_INPUT.out.diamond_nr_database_path
+        )
+        nt_full         = NUCLEOT_DIAMOND.out.reformed.map{it[1]}
+        nt_hits         = NUCLEOT_DIAMOND.out.hits_file.map{it[1]}
+        ch_versions     = ch_versions.mix(NUCLEOT_DIAMOND.out.versions)
+    } else {
+        nt_hits         = []
+        nt_full         = []
+    }
+
+    //
+    // SUBWORKFLOW: DIAMOND BLAST FOR INPUT ASSEMBLY
+    //
+    //qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore staxids sscinames sskingdoms sphylums salltitles
+    if ( include_workflow_steps.contains('uniprot_diamond') || include_workflow_steps.contains('ALL') ) {
+        UNIPROT_DIAMOND (
+            modified_input,
+            YAML_INPUT.out.diamond_uniprot_database_path
+        )
+        un_full         = UNIPROT_DIAMOND.out.reformed.map{it[1]}
+        un_hits         = UNIPROT_DIAMOND.out.hits_file.map{it[1]}
+        ch_versions     = ch_versions.mix(UNIPROT_DIAMOND.out.versions)
+    } else {
+        un_hits         = []
+        un_full         = []
+    }
+
+    ch_dot_genome = GENERATE_GENOME.out.dot_genome.map{it[1]}
+
+    CREATE_BTK_DATASET (
+        GENERATE_GENOME.out.reference_tuple,
+        ch_dot_genome,
+        ch_kmers,
+        ch_tiara,
+        ch_nt_blast,
+        ch_fcsgx,
+        ch_bam,
+        ch_coverage,
+        ch_kraken1,
+        ch_kraken2,
+        ch_kraken3,
+        nt_hits,
```

Comment:
I found what the bug was in the run on Friday: to get it to work, this here should be `nt_full` instead of `nt_hits` and on the line below, it should be `un_full` instead of `un_hits`

Reply:
Updated